### ISSUE TITLE
Macbook Pro 10,1 AHT not booting correctly

### ISSUE THREAD

**Issue by c0d3x** [upekkha/AppleHardwareTest#36]
I have a Macbook Pro A1387 (retina mid 2012) running latest OS Catalina 10.15.2 clean install.

I used this image: MacBookPro10,1 Mac-C3EC7CD22292981F
Which is the same as the one I got from the command in docs here.

How can I access AHT?

I made the bootable USB, and I get a yellow icon with EFI-BOOT.
Did not show up as AHT label.
I ran it and it flashed briefly some errors I could not read before it booted to OS.

**Comment by ryanbeymer** [comment]
Weird, I have the same hardware and just upgraded to Catalina. 

```
MacBookPro10,1
Mac-C3EC7CD22292981F
```

Started getting gpu kernel panics and started down the path of trying to run the diagnostics tests and ended up here. I'm experiencing the same issue, trying to run from usb and getting those quick errors.

**Comment by Treksten** [comment]
@ryanbeymer  I am in the same spot as you, same model and started getting GPU kernel panics, which seem to have begun after installing windows on bootcamp (win 10 1907), and i can't find out if hardware is the issue og software. Have you found a way to remove the panics, so the laptop won't shut down? Or maybe another community with a solution?

**Comment by ryanbeymer** [comment]
I was unable to find a solution, I ended up having my employer provide a new machine. 

I did read the real solution was to get a new gpu.

**Comment by Treksten** [comment]
That's what i thought, which doesn't really help me, since i am a really broke student.  But thanks for the answer

**Comment by TheRealXyzven** [comment]
Try to disable CSR like in this youtube video:

https://youtu.be/EEEIsnUKHVQ

**Comment by iminochkin** [comment]
> Try to disable CSR like in this youtube video:
> 
> https://youtu.be/EEEIsnUKHVQ
> 
> or this one
> 
> https://youtu.be/FxA5zamOvd4

Not working for me. MacBookPro10,1 Mac-C3EC7CD22292981F

**Comment by audiserg** [comment]
MacBookPro10,1 Mac-C3EC7CD22292981F also not work! I try many times from system disk  and from usb. Try 022-5882, and 3A244.

**Comment by ParkerRobb** [comment]
Tried the exact same thing on my MacBookPro10,1 and got the same results …

**Comment by TheRealXyzven** [comment]
Not sure why I never got notification about replies to this thread until ParkerRobb posted but here goes (a year later).  

I managed to run the AHT from the usb key after writing the AHT test onto it.  The only reason to boot into AHT (as far as I know) is to get a proper memory test done since once OSX/macOS is running, the AHT cannot do tests on those reserved memory address that OSX/macOS is using.  Of course (in my case), memory wasn't the issue for my mac, it was the actual GPU on my old MBP4,1 which has been known to have GPU BGA (ball grid array) issues with proper connection.

**Comment by sbarzaghialteaup** [comment]
MacBookPro10,1 Mac-C3EC7CD22292981F, some problem
I can see EFI-BOOT but AHT cannot start.
In the flash error messaging during start I have the message:
Cannot load "EFI/Drivers/TestSupport.efi"

**Comment by TheRealXyzven** [comment]
> MacBookPro10,1 Mac-C3EC7CD22292981F, some problem
> I can see EFI-BOOT but AHT cannot start.
> In the flash error messaging during start I have the message:
> Cannot load "EFI/Drivers/TestSupport.efi"

Yeah, I’m not sure why this is the case but you can rub AHT while booted in OS X provided you don’t need to run the memory test (per my last post).  So open that USB drive after OS X is booted.

**Comment by sbarzaghialteaup** [comment]
What do you mean for open the USB drive? Thanks

Scarica Outlook per Android<https://aka.ms/AAb9ysg>

________________________________
Da: TheRealXyzven ***@***.***>
Inviato: Domenica 5 Dicembre 2021, 01:28
A: upekkha/AppleHardwareTest
Cc: Samuele Barzaghi; Comment
Oggetto: Re: [upekkha/AppleHardwareTest] Macbook Pro 10,1 AHT not booting correctly (#36)


MacBookPro10,1 Mac-C3EC7CD22292981F, some problem
I can see EFI-BOOT but AHT cannot start.
In the flash error messaging during start I have the message:
Cannot load "EFI/Drivers/TestSupport.efi"

Yeah, I’m not sure why this is the case but you can rub AHT while booted in OS X provided you don’t need to run the memory test (per my last post). So open that USB drive after OS X is booted.

**Comment by TheRealXyzven** [comment]
Sorry, insert the USB drive after OS X has booted, you should be able to see the contents in it and run it. It’s been over 2 years since I did this so I can’t be certain what file to click to run the tests.
